docs/stack/client/howToUseClient.md

```diff
@@ -0,0 +1,203 @@
+
+  - _**contractInfos[]**_: contracts details. Can be set/fetch using subgraph by `setContractInfos`/`fetchContractInfos`
+  - _**graphqlHttpProvider**_: http connection to the subgraph of TheGraph protocol
+  - _**graphqlPrefetchHook()**_: function executed before sending graphQL query
+  - _**graphqlSubscribeToQueries**_: determines if query should subscribe to updates from the graphProvider. Default True
+  - _**graphqlWsProvider**_: web socket connection to subgraph of TheGraph protocol
+  - _**ipfsProvider**_: connection to ipfs provider which is used as the data storage layer by DAOstack
+  - _**web3Provider**_: connection to ethereum node
+  - _**web3ProviderRead**_: connection to ethereum node to read Arc data. If provided arc will read all data from this provider, else set same as `web3Provider`
+
+### Example
+
+```
+import { Arc } from '@daostack/client'
+
+const arc = new Arc({
+  graphqlHttpProvider: "https://subgraph.daostack.io/subgraphs/name/v23",
+  graphqlWsProvider: "wss://ws.subgraph.daostack.io/subgraphs/name/v23",
+  web3Provider: `wss://mainnet.infura.io/ws/v3/e0cdf3bfda9b468fa908aa6ab03d5ba2`,
+  ipfsProvider: {
+    "host": "subgraph.daostack.io",
+    "port": "443",
+    "protocol": "https",
+    "api-path": "/ipfs/api/v0/"
+  }
+})
+
+// before we can use the Arc instance to send transactions, we need to provide it
+// with information on where the contracts can be found
+// query the subgraph for the contract addresses, and use those
+await arc.fetchContractInfos()
+```
+
+**Note:**
+
+Some of these configuration settings are _**optional**_ to use `@daostack/client`. Like
+
+  - For creating and sending transactions to the blockchain, it is sufficient to provide the `web3Provider`.
+  - For fetching data from the subgraph, the `web3` and `ipfs` providers can be omitted when the library is only used.
+
+## Entities
+
+Entities are the basic building blocks of DAOstack ecosystem. All Entity classes can be created by providing an `id` (and an instance of `Arc`). 
+
+### Example:
+
+```
+const proposal = new Proposal('0x1234....', arc)
+```
+The proposal object can now be used to vote, and stake.
+```
+await proposal.vote(...).send()
+```
+This call will register a vote by sending a transaction to the blockchain. [See below](#sending-transactions) for details.
+
+Because the proposal is created with only an `id`, the client will query the subgraph for additional information, such as the address of the contract that the vote needs to be sent to. To make the client usable without having subgraph service available, all Entities have a second way of being created:
+```
+const proposal = new Proposal({
+  id: '0x12455..',
+  votingMachine: '0x1111..',
+  scheme: '0x12345...'
+}, arc)
+```
+This will provide the instance with enough information to send transactions without having to query the subgraph for additional information.
+
+
+### Common methods and properties
```

Comment:
we go to a lot of effort to support caching.  What is cached and how to we control it with respect to `search()`, `new` and `state()`?  At least refer to where this is more fully described.